### PR TITLE
Added filtering mechanism

### LINKED ISSUE #45: Ability to choose different game classes
Chess has game classes for all time controls.
For example:
180-599 seconds - blitz
600-1800 seconds - rapid
etc.

We can add dropdown which will sort data and show only specific game class chessheat.

In response from `https://api.chess.com/pub/player/{user}/games/{year}/{month}` we are getting `time_class` for every game.

### PR DESCRIPTION
**Pull request by e6voe9**
Now we can filter heatmap by time classes: 
- bullet
- blitz
- rapid
- dayli

Filter is inactive when no user selected. Filter also becomes inactive on submit error.

Filter resets when you submit form.

Wasn't working on styles much for now.

Refactored a little bit, to make this work.
`fetchData` now only fetching data. 
Extracted filling table with data logic to `fillTableWithData` function.
Added `runHeatMap` function.

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -113,9 +133,11 @@ function setHue() {
 /* End hue slider logic */
 
 generateTable();
+FilterOptions.generate();
+FilterOptions.disable();
 queryBasedOnQueryParams();
 
-function queryBasedOnQueryParams() {
+async function queryBasedOnQueryParams() {
```
Comment (maxdemaio) [inline script.js:140]: I think we could keep this as synchronous and rm `async` 👍🏻

**Review by maxdemaio — APPROVED** [review]
Everything is looking good to me! With the introduction of async/await we will be able to add try / catch blocks around the asynchronous code. Glad you added the `time-classes.js` file. I plan to do some similar changes to help break up how large `script.js` is 👍🏻

The only thing I could think of is we could use the native `select` element in HTML. Since we know these won't change maybe something like this:

```html
<div class="filter">
  <select name="filter" id="filter-select">
    <option value="">any</option>
    <option value="bullet">bullet</option>
    <option value="blitz">blitz</option>
    <option value="rapid">rapid</option>
    <option value="daily">daily</option>
  </select>
</div>
```

**Comment by e6voe9** [comment]
Ok, I made it in javascript because wanted to keep all options logic together in one place.
Feel free to place it right in the HTML if it's necessary 🙂. Then we don't need `generate` function any more.

**Comment by maxdemaio** [comment]
> Ok, I made it in javascript because wanted to keep all options logic together in one place. Feel free to place it right in the HTML if it's necessary 🙂.

Lol nope it's not really necessary just the only thing I could find 😆 I'm going to merge this in!

**Pull request merged**